go/vt/vtgate/planbuilder/selectGen4.go

```diff
@@ -218,22 +354,23 @@ func needsWeightString(tbl semantics.TableInfo, colName *sqlparser.ColName) bool
 	return true // we didn't find the column. better to add just to be safe1
 }
 
-func wrapExprAndPush(exp sqlparser.Expr, plan logicalPlan, semTable *semantics.SemTable) (int, error) {
-	aliasedExpr := &sqlparser.AliasedExpr{Expr: exp}
-	offset, _, err := pushProjection(aliasedExpr, plan, semTable, true)
-	return offset, err
-}
-
-func planOrderByForJoin(qp *queryProjection, orderExprs []orderBy, plan *joinGen4, semTable *semantics.SemTable) (logicalPlan, error) {
+func planOrderByForJoin(qp *abstract.QueryProjection, orderExprs []abstract.OrderBy, plan *joinGen4, semTable *semantics.SemTable) (logicalPlan, bool, error) {
 	if allLeft(orderExprs, semTable, plan.Left.ContainsTables()) {
-		newLeft, err := planOrderBy(qp, orderExprs, plan.Left, semTable)
+		newLeft, _, err := planOrderBy(qp, orderExprs, plan.Left, semTable)
 		if err != nil {
-			return nil, err
+			return nil, false, err
 		}
 		plan.Left = newLeft
-		return plan, nil
+		return plan, false, nil
 	}
+	ms, colAdded, err := createMemorySortPlan(plan, orderExprs, semTable)
```

Comment:
we could just return here, no need to check `err`

Reply:
Resolved via 03aa03adb0